src/test/java/com/puppycrawl/tools/checkstyle/TreeWalkerTest.java

```diff
@@ -93,18 +92,22 @@ protected String getPackageLocation() {
 
     @Test
     public void testProperFileExtension() throws Exception {
-        final DefaultConfiguration checkConfig =
-                createModuleConfig(ConstantNameCheck.class);
         final File file = new File(temporaryFolder, "file.java");
         try (Writer writer = Files.newBufferedWriter(file.toPath(), StandardCharsets.UTF_8)) {
-            final String content = "public class Main { public static final int k = 5 + 4; }";
+            final String content = "/*\n"
+                    + "com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck\n\n"
+                    + "*/\n"
+                    + "package com.puppycrawl.tools.checkstyle.TreeWalkerTest;\n\n"
+                    + "public class Main { \n\n"
+                    + "  //violation below\n"
+                    + "  public static final int k = 5 + 4;  \n}";
             writer.write(content);
         }
         final String[] expected1 = {
-            "1:45: " + getCheckMessage(ConstantNameCheck.class,
-                    MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"),
+            "10:27: " + getCheckMessage(ConstantNameCheck.class,
+                        MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"),
         };
-        verify(checkConfig, file.getPath(), expected1);
+        verifyWithInlineConfigParser(file.getPath(), expected1);
```

Comment:
```suggestion
    @Test
    public void testProperFileExtension() throws Exception {
        final String path = getPath("InputProperFileExtension.java");
        final String[] expected = {
            "10:27: " + getCheckMessage(ConstantNameCheck.class,
                        MSG_INVALID_PATTERN, "k", "^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$"),
        };
        verifyWithInlineConfigParser(path, expected);
    }

```

Where `InputProperFileExtension.java` is in the folder specified in the `getPackageLocation()` method above, and looks like this:
```java
/*
* com.puppycrawl.tools.checkstyle.checks.naming.ConstantNameCheck
*/

package com.puppycrawl.tools.checkstyle.treewalker;

public class InputProperFileExtension {

    public static final int k = 5 + 4;
    // violation above
}


```

Reply:
This is an example of item (2) from https://github.com/checkstyle/checkstyle/pull/14388#issuecomment-1918347975.

Reply:
AllTestsTest.testAllInputsHaveTest:60->walk:87->lambda$testAllInputsHaveTest$1:61->verifyInputFile:170->verifyInputFile:202 Resource must be named after a Test like 'InputMyCustomCase.java' and be in the sub-package of the test like 'mycustom' for test 'MyCustomCheckTest': com\puppycrawl\tools\checkstyle\treewalker\InputProperFileExtension.java
expected to be true

I got this error after making this change

Reply:
Try to rename file like `InputTreewalkerProperFileExtension`

Reply:
@nrmancuso Yes exactly. I renamed it and it worked